apps/legal_archive/commands.py

```diff
@@ -332,12 +332,12 @@ class ImportLegalPublishQueueCommand(superdesk.Command):
     def run(self):
         logger.info('Import to Legal Publish Queue')
         lock_name = get_lock_id('legal_archive', 'import_legal_publish_queue')
-        if not lock(lock_name, '', expire=600):
+        if not lock(lock_name, 'superdesk', expire=600):
             return
         try:
             LegalArchiveImport().import_legal_publish_queue()
         finally:
-            unlock(lock_name, '')
+            unlock(lock_name, 'superdesk')
```

Comment:
imo this can make the issue worse, because every worker can unlock it for every other worker. so if worker A starts and it doesn't finish before expiry time, worker B will get the lock, and then worker A finishes and will unlock it. so any other worker after can get lock even if worker B is still working. and B will unlock it when finished, no matter if it got picked by other workes. etc..